### PR TITLE
Add a section on referencing or embedding external credentials.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Addresses issue #4.

***
<a href="https://pr-preview.s3.amazonaws.com/dmitrizagidulin/vc-imp-guide/pull/51.html" title="Last updated on Aug 27, 2019, 2:44 PM UTC (22cf25b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-imp-guide/51/950a115...dmitrizagidulin:22cf25b.html" title="Last updated on Aug 27, 2019, 2:44 PM UTC (22cf25b)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -246,6 +248,68 @@ <h4>Verifiable Data Registries</h4>
...
+Usage of <a>verifiable credentials</a> will often require referencing other
+credentials, embedding or attaching multiple credentials, or otherwise binding
+them together.
+      </p>
+
+      <section>
+        <h4>Referencing Credentials Without Integrity Protection</h4>
+        <p>
+The simplest way for a <a>credential</a> to reference another external
+credential is to link to it, either explicitly by using a URI, or implicitly by
```
Comment (David-Chadwick) [inline index.html]: is to explicitly link to it, either directly by using its URI, or indirectly by

File: index.html [hunk 2]
```diff
@@ -246,6 +248,68 @@ <h4>Verifiable Data Registries</h4>
...
+providing a well-known ID (for example, a credential modeling an internal
+company Invoice may refer to its parent Purchase Order credential simply by the
+PO Number, relevant only within the context of this specific enterprise).
+        </p>
+
+        <p>
+This method of linking to an external credential without using an integrity
+protection mechanism may be acceptable in rare use cases, such as when both
```
Comment (David-Chadwick) [inline index.html]: Delete "rare" since this will be a common use case. Replace with "some"

File: index.html [hunk 3]
```diff
@@ -246,6 +248,68 @@ <h4>Verifiable Data Registries</h4>
...
+credentials are issued by the same entity, the verifier has a high level
+of trust and confidence in the issuer's security and auditing mechanisms, and
+the value of the credentials is sufficiently low. However, this is not
+recommended, and goes against the philosophy of <a>verifiable credentials</a>.
```
Comment (David-Chadwick) [inline index.html]: I disagree. It does not go against the philosophy of VCs. Please delete the whole section starting with However, this is not.....
The referenced credential may be just as valid as the referencing credential since tampering with either will normally be just as difficult for an attacker.

File: index.html [hunk 4]
```diff
@@ -246,6 +248,68 @@ <h4>Verifiable Data Registries</h4>
...
+      </section>
+
+      <section>
+        <h4>Referencing Credentials With Integrity Protection</h4>
+        <p>
+The recommended way of referencing an external credential from within a
+<a>verifiable credential</a> is to use a linking mechanism that
+cryptographically binds the contents of the target document to the URI itself.
+One way to accomplish this would be to use <a>hashlinks</a> or an equivalent
+URI scheme. Another mechanism would be to encode the full contents of the
```
Comment (David-Chadwick) [inline index.html:291]: I dont see the point in encoding the full contents, especially if the issuer is the same for both credentials. I suggest deleting the section starting Another mechanism

**Review by David-Chadwick — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (dmitrizagidulin) [reply index.html]: Changed, thanks.

File: index.html (hunk 2 above)
Comment (dmitrizagidulin) [reply index.html]: Replaced with "some".

File: index.html (hunk 3 above)
Comment (dmitrizagidulin) [reply index.html]: A large part of the purpose of Verifiable Credentials (the verifiable part) is that they contain tamper-evident mechanisms like digital signatures. 
The thing about linking to external credentials is that the digital signature on the content of one of them (without an equivalent mechanism in the link itself) will give users a fall sense of security, since they may assume (incorrectly) that the signature also somehow binds or protects the content of the other credential.

To put it another way, if you're not cryptographically binding the contents of the other credential, why use a verifiable credential in the first place? Why not use a regular one?

The example given of just simply using a foreign key (like referencing a PO Number for an Invoice credential), that's very commonly done, I agree. But it's done in use cases where there's a very low threat of tampering, in which case, why use a verifiable credential at all?

**Review by TallTed — APPROVED** [review]

File: index.html [hunk 5]
```diff
@@ -261,24 +261,23 @@ <h3>Referencing Other Credentials</h3>
...
-recommended, and goes against the philosophy of <a>verifiable credentials</a>.
-Although the credential that contains the reference may be integrity protected
-itself (by a cryptographic signature or a similar proof mechanism), the verifier
-has no way of knowing that the external credential being linked to has not been
-tampered with, unless the link itself has a content integrity protection
-mechanism built into it.
+protection mechanism may be acceptable in some use cases, such as when both
+credentials are issued by the same entity, the verifier has a high level of
+trust and confidence in the issuer's security and auditing mechanisms, and the
+value of the credentials is sufficiently low. However, implementers should keep
```
Comment (David-Chadwick) [inline index.html]: risk to the verifier is sufficiently low.

Comment (dmitrizagidulin) [reply index.html]: Rephrased, thanks!

**Review by David-Chadwick — APPROVED** [review]
